### PR TITLE
fix(core): field ui-options higher priority

### PR DESCRIPTION
**Pull request by shaddollxz**
### Reasons for making this change

[Please describe them here]

If this is related to existing tickets, include links to them as well. Use the syntax `fixes #[issue number]` (ex: `fixes #123`).

If your PR is non-trivial and you'd like to schedule a synchronous review, please add it to the weekly meeting agenda: https://docs.google.com/document/d/12PjTvv21k6LIky6bNQVnsplMLLnmEuypTLQF8a-8Wss/edit

### Checklist

- [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [x] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by shaddollxz** [comment]
https://github.com/rjsf-team/react-jsonschema-form/issues/4210

**Comment by heath-freenome** [comment]
@shaddollxz Your tests aren't passing

**Comment by shaddollxz** [comment]
> @shaddollxz Your tests aren't passing

Solved, I missed modifying some types

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ should change the heading of the (upcoming) version to include a major version b
 
 - Fix case where NumberField would not properly reset the field when using programmatic form reset (#4202)[https://github.com/rjsf-team/react-jsonschema-form/issues/4202]
 - Updated widgets to handle undefined `target` in `onFocus` and `onBlur` handlers
+- Fix field disable or readonly property cant cover globalOptions corresponding property (#4212)[https://github.com/rjsf-team/react-jsonschema-form/pull/4212]
```
Comment (heath-freenome) [inline CHANGELOG.md:37]:
```suggestion
- Fix field disable or readonly property can't cover globalOptions corresponding property (#4212)[https://github.com/rjsf-team/react-jsonschema-form/pull/4212]
```

**Review by heath-freenome — APPROVED** [review]

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**